sql/src/main/java/org/apache/druid/sql/calcite/rule/FilterJoinExcludePushToChildRule.java

```diff
@@ -0,0 +1,277 @@
+  /** Copied from {@link FilterJoinRule#NOT_ENUMERABLE} */
+  private static final Predicate NOT_ENUMERABLE = (join, joinType, exp) ->
+      join.getConvention() != EnumerableConvention.INSTANCE;
+
+  /**
+   * Rule that pushes predicates from a Filter into the Join below them.
+   * Similar to {@link FilterJoinRule#FILTER_ON_JOIN} but does not push predicate to the child
+   */
+  public static final FilterJoinRule FILTER_ON_JOIN_EXCLUDE_PUSH_TO_CHILD =
+      new FilterIntoJoinExcludePushToChildRule(RelFactories.LOGICAL_BUILDER, NOT_ENUMERABLE);
+
+  FilterJoinExcludePushToChildRule(RelOptRuleOperand operand,
+                                             String id,
+                                             boolean smart,
+                                             RelBuilderFactory relBuilderFactory,
+                                             Predicate predicate) {
+    super(operand, id, smart, relBuilderFactory, predicate);
+  }
+
+  /**
+   * Rule that tries to push filter expressions into a join
+   * condition. Exlucde pushing into the inputs (child) of the join.
+   */
+  public static class FilterIntoJoinExcludePushToChildRule extends FilterJoinExcludePushToChildRule
+  {
+    public FilterIntoJoinExcludePushToChildRule(RelBuilderFactory relBuilderFactory, Predicate predicate) {
+      super(
+          operand(Filter.class,
+                  operand(Join.class, RelOptRule.any())),
+          "FilterJoinExcludePushToChildRule:filter", true, relBuilderFactory,
+          predicate);
+    }
+
+    @Override
+    public void onMatch(RelOptRuleCall call) {
+      Filter filter = call.rel(0);
+      Join join = call.rel(1);
+      perform(call, filter, join);
+    }
+  }
+
+  /**
+   * Copied from {@link FilterJoinRule#perform}
+   * The difference is that this method will not not push filters to the children in classifyFilters
+   */
+  @Override
+  protected void perform(RelOptRuleCall call, Filter filter,
+                         Join join) {
+    final List<RexNode> joinFilters =
+        RelOptUtil.conjunctions(join.getCondition());
+    final List<RexNode> origJoinFilters = ImmutableList.copyOf(joinFilters);
+    // If there is only the joinRel,
+    // make sure it does not match a cartesian product joinRel
+    // (with "true" condition), otherwise this rule will be applied
+    // again on the new cartesian product joinRel.
+    if (filter == null && joinFilters.isEmpty()) {
+      return;
+    }
+
+    final List<RexNode> aboveFilters =
+        filter != null
+        ? getConjunctions(filter)
+        : new ArrayList<>();
+    final ImmutableList<RexNode> origAboveFilters =
+        ImmutableList.copyOf(aboveFilters);
+
+    // Simplify Outer Joins
+    JoinRelType joinType = join.getJoinType();
+    if (!origAboveFilters.isEmpty() && join.getJoinType() != JoinRelType.INNER) {
+      joinType = RelOptUtil.simplifyJoin(join, origAboveFilters, joinType);
+    }
+
+    final List<RexNode> leftFilters = new ArrayList<>();
```

Comment:
These left right thing is not needed. But I copied this whole code from Calcite and made minimum change (which is just the classifyFilters method). Hence, keeping these variables here as I don't want to refactor Calcite stuff more than needed.